mm2src/coins/utxo/utxo_block_header_storage/sql_block_header_storage.rs

```diff
@@ -225,26 +242,25 @@ impl BlockHeaderStorageOps for SqliteBlockHeadersStorage {
         })
     }
 
-    async fn get_last_block_height(&self) -> Result<u64, BlockHeaderStorageError> {
+    async fn get_last_block_height(&self) -> Result<Option<u64>, BlockHeaderStorageError> {
         let coin = self.ticker.clone();
         let sql = get_last_block_height_sql(&coin)?;
         let selfi = self.clone();
 
         async_blocking(move || {
             let conn = selfi.conn.lock().unwrap();
-            query_single_row(&conn, &sql, NO_PARAMS, |row| row.get(0))
+            if let Some(height) = query_single_row(&conn, &sql, NO_PARAMS, |row| row.get::<usize, i64>(0))? {
+                let height = height
+                    .try_into()
+                    .map_err(|e: TryFromIntError| BlockHeaderStorageError::DecodeError {
+                        coin,
+                        reason: e.to_string(),
+                    })?;
+                return Ok(Some(height));
+            };
+            Ok(None)
```

Comment:
`query_single_row` returns an option so this can be written as
```rust
let maybe_height = async_blocking(move || {
    let conn = selfi.conn.lock().unwrap();
    query_single_row(&conn, &sql, NO_PARAMS, |row| row.get::<_, i64>(0))
})
.await
.map_err(|e| BlockHeaderStorageError::GetFromStorageError {
    coin: coin.clone(),
    reason: e.to_string(),
})?;
maybe_height
    .map(|h| h.try_into())
    .transpose()
    .map_err(|e: TryFromIntError| BlockHeaderStorageError::DecodeError {
        coin,
        reason: e.to_string(),
    })
```
Please note that mapping to `GetFromStorageError` will show both the coin and the query in the error, so it's good to keep it.

Reply:
ok